MyFirstSlideshow/app/src/main/AndroidManifest.xml

```diff
@@ -2,6 +2,12 @@
 <manifest package="com.charles.myfirstslideshow"
     xmlns:android="http://schemas.android.com/apk/res/android">
 
+    <uses-permission android:name="android.permission.INTERNET" />
+    <uses-permission android:name="android.permission.ACCESS_WIFI_STATE" />
+    <uses-permission android:name="android.permission.CHANGE_WIFI_STATE" />
+    <uses-permission android:name="android.permission.CHANGE_NETWORK_STATE" />
```

Comment:
Do you really need all of these permissions? Why do you need to see the "CHANGE_WIFI_STATE"? If you don't need some of them, please remove it. 
See the documentation to understand what permissions are really needed.